### PR TITLE
Initial ansible-test sanity docs.

### PR DESCRIPTION
**Pull request by mattclay**
##### SUMMARY

Initial ansible-test sanity docs:

- Add a dedicated documentation page for each `ansible-test sanity` command.
- Enforce the presence of documentation for each command in the sanity tests.

Once this documentation is deployed at a permanent location, we can do the following:

- Update ansible-test to include a docs URL for each bot test result file.
- Update the bot to include a link to error-specific docs in PR comments.

This should make the bot comments with error messages much more useful to contributors, as there will be documentation available for any given category of test. This PR provides only minimal documentation to get things started. Once deployed the documentation can be expanded to include the rationale for each test as well as detailed instructions on how to correct reported issues.

##### ISSUE TYPE

Feature Pull Request

##### COMPONENT NAME

ansible-test

##### ANSIBLE VERSION

```
ansible 2.4.0 (at-sanity-docs 5eba80e615) last updated 2017/07/13 18:37:49 (GMT -700)
  config file = None
  configured module search path = [u'/Users/mclay/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/mclay/code/mattclay/ansible/lib/ansible
  executable location = /Users/mclay/code/mattclay/ansible/bin/ansible
  python version = 2.7.11 (default, Jan 22 2016, 08:29:18) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

### CODE REVIEW

**Comment by mattclay** [comment]
We need to make sure that before we start linking to these pages that the location will not change.

File: docs/docsite/rst/dev_guide/testing/sanity/no-list-cmp.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Sanity Tests » no-list-cmp
+==========================
+
+The ``cmp`` function has been removed in python 3.
```
Comment (dharmabumstead) [inline docs/docsite/rst/dev_guide/testing/sanity/no-list-cmp.rst]: All instances of the word "Python" that aren't in a code block should be capitalized.

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
Looks great, except 'python' should be capitalized.

**Review by dharmabumstead — APPROVED** [review]
Thanks Matt!

**Comment by mattclay** [comment]
If we want to rename any of the sanity tests, now is the time to do it. Once this is deployed we should avoid renaming them. Otherwise we'll have to create redirection pages when renaming tests or have broken links from older PRs.

**Comment by mattclay** [comment]
I believe we'll need a temporary redirect until the [versioned docs](https://github.com/ansible/proposals/issues/60) are in place for:

`/ansible/devel/dev_guide/testing/sanity/` -> `/ansible/dev_guide/testing/sanity/`

That way the links will continue to work once we deploy versioned docs.

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Looks good.
Let's get this in then we can iterate on the individual files.

**Pull request merged**